### ISSUE TITLE
Site Assembler: Always focus on an element when switching the UI

### ISSUE THREAD

**Issue by okmttdhr** [Automattic/wp-calypso#74189]
### What

It would be helpful to decide which element to focus on first, even after the screen/UI has switched. 

Current behavior (no focusing after switching the UI);

https://user-images.githubusercontent.com/5287479/223656442-f9f1e466-d0a9-407f-b2e8-4e5bb2188667.mov

For example, the focus is on the back button on the site editor page. It would be better to focus on the element you were focusing on before switching the UI (we could discuss it).

https://user-images.githubusercontent.com/5287479/223656740-88931cef-565d-4c4f-9408-9a8c61f5dfbf.mov

### Why

For keyboard accessibility friendly UI 🙂 

### Related

- https://github.com/Automattic/wp-calypso/pull/74157#issuecomment-1459578432
- https://github.com/Automattic/wp-calypso/issues/74188

**Comment by arthur791004** [comment]
The `NavigatorProvider` did this but there is an issue with our use case. For example, when the element you were focusing on before switching the UI is **gone** (e.g. Remove), it will focus on the first focusable element (e.g. Move down) but it looks a bit weird (at least for me 😂). That's why I disabled the [focus restoration](https://github.com/Automattic/wp-calypso/blob/f7f9f6e82bfe92be14d54faae2162e1895e162d0/client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx#L487) temporarily

**Comment by okmttdhr** [comment]
I see, I understand the background. Thank you for sharing!

**Comment by arthur791004** [comment]
Closing it via https://github.com/Automattic/wp-calypso/pull/75948
